### ISSUE TITLE
Socket closed while waiting on TCP (minecraft)

### ISSUE THREAD

**Issue by Murgator** [sonicsnes/node-gamedig#117]
Hi ! I have a problem with this minecraft server: play.mcroyal.fr:25565
I obtain this error with the old version of gamedig: Socket closed while waiting on TCP

And this error with the last version of gamedig:

![61f5edb3603ed17aa6b6c6d21d003f92](https://user-images.githubusercontent.com/38978895/53878422-1e152c00-400c-11e9-8019-8716efa6962c.png)

But this server is online ! Try here: https://dinnerbone.com/minecraft/tools/status/

I obtain the same errors in my local computer and on my vps.

Thank you in advance
David

**Comment by robbiek** [comment]
Cant help you in the issue but a temporary fix for this is to use minecraftpe instead of minecraft to query the server. 

this is the response i get back using minecraftpe

`{"name":"§6§lMCROYAL » §eLe Meilleur serveur Farm2Win !\n§6Pvp §ff §6Faction §ff §6KoTH & Totems §ff §6Events réguliers","map":"","password":false,"raw":{"playerTeamInfo":{"":[{"player":"Patheoche"},{"player":"13chrism"},{"player":"Raven"},{"player":"siftix_17"},{"player":"Stonik"},{"player":"Neydo"},{"player":"BryanYT"},{"player":"Dark_UniCron"},{"player":"Allwords"},{"player":"Realexx_"},{"player":"Petitpomme"},{"player":"DJPonnet"},{"player":"nathancremeu"},{"player":"StrongBoy999"},{"player":"Trefle16"},{"player":"Lxgends"},{"player":"toto"},{"player":"stilgarmat"},{"player":"14kaara"},{"player":"olivier"},{"player":"zeliote"},{"player":"TheSupreme"},{"player":"FurixEnigma"},{"player":"msgf"},{"player":"ShrawKs"},{"player":"sebstone191"},{"player":"BrangHD"},{"player":"Jack_Daniels"},{"player":"isolender"},{"player":"St4linator"},{"player":"Ewans131"},{"player":"NeeDox_"},{"player":"PNLItachi23"}]},"hostname":"§6§lMCROYAL » §eLe Meilleur serveur Farm2Win !\n§6Pvp §ff §6Faction §ff §6KoTH & Totems §ff §6Events réguliers","gametype":"SMP","game_id":"MINECRAFT","version":"1.8","plugins":"","map":"Waterfall_Proxy","numplayers":"33","maxplayers":"150","hostport":"25565","hostip":"0.0.0.0"},"maxplayers":150,"players":[{"name":"Patheoche"},{"name":"13chrism"},{"name":"Raven"},{"name":"siftix_17"},{"name":"Stonik"},{"name":"Neydo"},{"name":"BryanYT"},{"name":"Dark_UniCron"},{"name":"Allwords"},{"name":"Realexx_"},{"name":"Petitpomme"},{"name":"DJPonnet"},{"name":"nathancremeu"},{"name":"StrongBoy999"},{"name":"Trefle16"},{"name":"Lxgends"},{"name":"toto"},{"name":"stilgarmat"},{"name":"14kaara"},{"name":"olivier"},{"name":"zeliote"},{"name":"TheSupreme"},{"name":"FurixEnigma"},{"name":"msgf"},{"name":"ShrawKs"},{"name":"sebstone191"},{"name":"BrangHD"},{"name":"Jack_Daniels"},{"name":"isolender"},{"name":"St4linator"},{"name":"Ewans131"},{"name":"NeeDox_"},{"name":"PNLItachi23"}],"bots":[],"connect":"play.mcroyal.fr:25565","ping":39}`

**Comment by Murgator** [comment]
Thank you robbiek i use minecraftpe waiting for an update of gamedig.

**Comment by sonicsnes** [comment]
play.mcroyal.fr is a Waterfall Proxy, not a single minecraft server. It appears that Waterfall and Bungeecord have decided to implement the minecraft bedrock edition query protocol (gamespy3) rather than the standard minecraft query protocol. This is strange, but whatever. I suppose maybe we should just try both for all minecraft queries?

**Comment by sooxiaotong** [comment]
I had changed to minecraftpe, it also doesn't show players.

![image](https://user-images.githubusercontent.com/13378059/54078947-3a1b1500-430d-11e9-92f3-36e18aa485d1.png)

**Comment by sonicsnes** [comment]
It looks like there was actually nobody online when you queried it. I just tried and it seems to work with minecraftpe:

```
{
  "name": "§6§lMCROYAL » §eLe Meilleur serveur Farm2Win !\n§6Pvp §ff §6Faction §ff §6KoTH & Totems §ff §6Events réguliers",
  "map": "",
  "password": false,
  "raw": {
    "playerTeamInfo": {
      "": [
        {
          "player": "Minokoala"
        },
        {
          "player": "siftix_17"
        },
        {
          "player": "zLukoo"
        },
        {
          "player": "isolender"
        },
        {
          "player": "exogame17"
        },
        {
          "player": "Neydo"
        },
        {
          "player": "___NaRuTo"
        }
      ]
    },
    "hostname": "§6§lMCROYAL » §eLe Meilleur serveur Farm2Win !\n§6Pvp §ff §6Faction §ff §6KoTH & Totems §ff §6Events réguliers",
    "gametype": "SMP",
    "game_id": "MINECRAFT",
    "version": "1.8",
    "plugins": "",
    "map": "Waterfall_Proxy",
    "numplayers": "7",
    "maxplayers": "150",
    "hostport": "25565",
    "hostip": "0.0.0.0"
  },
  "maxplayers": 150,
  "players": [
    {
      "name": "Minokoala"
    },
    {
      "name": "siftix_17"
    },
    {
      "name": "zLukoo"
    },
    {
      "name": "isolender"
    },
    {
      "name": "exogame17"
    },
    {
      "name": "Neydo"
    },
    {
      "name": "___NaRuTo"
    }
  ],
  "bots": [],
  "connect": "play.mcroyal.fr:25565",
  "ping": 145
}
```

**Comment by Murgator** [comment]
I confirm that minecraftpe works with the players list.

**Comment by sooxiaotong** [comment]
@Murgator @sonicsnes Heres my script that wont work

```
const Discord = require('discord.io');
const logger = require('winston');
const auth = require('./auth.json');
const request = require('request');
const Gamedig = require('gamedig');
const Sequelize = require('sequelize');

const sequelize = new Sequelize('authme', 'authme', '12345', {
	host: 'localhost',
	dialect: 'mysql',
});

const MCQ = {
	type: 'minecraftpe',
	host: '10.106.50.5',
    port: 25565
}

//Creeper behavior
var sleep = false;
var gunpowder_count = 0;
const EXPLOSION_MESSAGES = {
	3:'Feed me more gunpowder...',
	6:'Pop!',
	9:'Boom!',
	12:'BOOOOM!!!',
	15:'KaBOOOM!!!',
	18:'KaBoOoOoOoMMM!!!!!',
	45:'That\'s some... dedications...',
	90:'Are you serious? What kind of sounds I can make at this rate...'
};

// Configure logger settings
logger.remove(logger.transports.Console);
logger.add(new logger.transports.Console, {
    colorize: true
});
logger.level = 'debug';
// Initialize Discord Bot
var bot = new Discord.Client({
   token: auth.token,
   autorun: true,
   autoReconnect:true
});
bot.on('ready', function (evt) {
    logger.info('Connected');
    logger.info('Logged in as: ');
    logger.info(bot.username + ' - (' + bot.id + ')');
});
bot.on('message', async function (user, userID, channelID, message, evt) {
    // Our bot needs to know if it will execute a command
    // It will listen for messages that will start with `/`
    if (message.substring(0, 1) == '/') {
        var args = message.substring(1).split(' ');
        var cmd = args[0];
       
        args = args.splice(1);
        switch(cmd) {
            // !ping
            case 'ping':
                bot.sendMessage({
                    to: channelID,
                    message: 'Psssss...'
                });
            break;

            case 'server':
                const state1 = await Gamedig.query(MCQ);
                bot.sendMessage({
                    to: channelID,
                    message: `\`\`\`${JSON.stringify(state1, null, "  ")} \`\`\``
                    //message: `\`\`\`== Online Players [${state.players.length}] == \n${state.players.map(p => p.name + "\n").join('')}\`\`\``
                });
            break;

            case 'online':
                const state = await Gamedig.query(MCQ);
                bot.sendMessage({
                    to: channelID,
                    //message: `\`\`\`${JSON.stringify(state, null, "  ")} \`\`\``
                    message: `\`\`\`== Online Players [${state.players.length}] == \n${state.players.map(p => p.name + "\n").join('')}\`\`\``
                });
			break;
			
			case 'where':
				//Dev role ID: 501413758942576643
				if (!(evt.d.member.roles.indexOf('501413758942576643') > -1)) {
					break;
				}
				if (args.length > 1) {
					bot.sendMessage({
						to: userID,
						message: 'I am confused... °‿‿°'
					});
					break;
				}
				let uname = args[0];
				sequelize
					.query(
						'SELECT x,y,z FROM authme WHERE username = ?',
						{ raw: true, replacements: [uname] }
					)
					.spread((results, metadata) => {
						// Raw query - use spread
						if(results.length)
							msg = `${uname} is at ${results[0].x}, ${results[0].y}, ${results[0].z}`
						else
							msg = 'Poofs! Player missing.'
						bot.sendMessage({
							to: userID,
							message: `\`\`\`${msg}\`\`\``
						});
					});
            break;

            case 'give':
            	if (args.length > 1) {
            		bot.sendMessage({
            			to: channelID,
            			message: 'I am confused... °‿‿°'
            		});
            		break;
            	}
            	let givenItem = args[0];
            	switch(givenItem)
            	{
            		case 'gunpowder':{
            			const MESSAGE = [
            				'Taste good!',
            				'I am growing up.',
            				'I am getting bigger.',
            				'I want more!'
            			];

            			gunpowder_count += 1;
            			bot.sendMessage({
            				to: channelID,
            				message: MESSAGE[Math.floor(Math.random()*MESSAGE.length)]
            			});
            		break;
            		}

            		case 'ice':{
            			const MESSAGE = [
            				'Feels like sleeping...',
            				'I am cooling down...',
            				'Feels hungry :(',
            				'It\'s so cold...'
            			];
            			if (gunpowder_count > 0){
            				gunpowder_count -= 1;
            			}

            			bot.sendMessage({
            				to: channelID,
            				message: MESSAGE[Math.floor(Math.random()*MESSAGE.length)]
            			});
            		break;
            		}

            		case 'apple':{
            			const state = await Gamedig.query(MCQ);
            			if (state.players.length <= 0)
            			{
            				bot.sendMessage({
            					to: channelID,
            					message: 'No one is here...'
            				});
            				break;
            			}
            			bot.sendMessage({
            				to: channelID,
            				message: 'Time to explode infront of **' + state.players[Math.floor(Math.random()*state.players.length)].name + '**!'
            			});
            		break;
            		}

            		case 'diamond':
            			bot.sendMessage({
            				to: channelID,
            				message: 'I\'ll give you back: ' + Math.ceil(Math.random()*100) + ' emerald(s)!'
            			});
            		break;

            		case 'torch':
            			bot.sendMessage({
            				to: channelID,
            				message: 'Heat me up, I am at level: ' + gunpowder_count
            			});
            		break;

            		default:
            			const GIVABLE_ITEMS = [
            				'-',
            				'gunpowder - Charge up the Creeper bot.',
            				'ice - Cool down the Creeper bot.',
            				'apple - Selects a random online player.',
            				'diamond - Give you back randomly from 1 to 100 Emeralds.',
            				'torch - Shows the charge level.'
            			];
            			bot.sendMessage({
            				to: channelID,
            				message: 'Here are a list of givable items\`\`\`'+GIVABLE_ITEMS.join('\n')+'\`\`\`'
            			});
            		break;
            	}
            break;

            case 'explode':
            	for (var value in EXPLOSION_MESSAGES){
            		if (gunpowder_count <= value){
            			gunpowder_count = 0;
            			bot.sendMessage({
            				to: channelID,
            				message: EXPLOSION_MESSAGES[value]
            			});
            			break;
            		}
            	}
            break;

            case 'help':
            	const COMMAND_LIST = [
            		'ping - To make sure Creeper bot is not sleeping.',
            		'online - List every online players name.',
            		'give - Give something to Creeper!',
            		'explode - Detonate the Creeper :\'('
            	];
            	bot.sendMessage({
            		to: channelID,
            		message: 'Available commands\`\`\`' + COMMAND_LIST.join('\n') + '\`\`\`'
            	});
            break;


            // Just add any case commands if you want to..
         }
     }
});

```

**Comment by Murgator** [comment]
This server not working with minecraftpe and minecraftping protocol: play.skaia.us

minecraftping:
![minecraftping](https://user-images.githubusercontent.com/38978895/54355667-f3edea80-4659-11e9-9baf-5c4e3ec06211.png)

minecraftpe:
![minecraftpe](https://user-images.githubusercontent.com/38978895/54355678-fb14f880-4659-11e9-932e-9779e86764f2.png)

https://dinnerbone.com/minecraft/tools/status/ say: This server is hiding their player list. But is online.

**Comment by sonicsnes** [comment]
Closing as a duplicate of #116
